Core/include/Acts/EventData/SpacePointContainer.ipp

```diff
@@ -0,0 +1,271 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <cmath>
+
+namespace Acts {
+
+template <typename container_t, template <typename> class holder_t>
+template <template <typename> class, typename>
+SpacePointContainer<container_t, holder_t>::SpacePointContainer(
+    const Acts::SpacePointContainerConfig& config,
+    const Acts::SpacePointContainerOptions& options, container_t& container)
+    : m_config(config.toInternalUnits()),
+      m_options(options.toInternalUnits()),
+      m_container(container) {
+  this->initialize();
+}
+
+template <typename container_t, template <typename> class holder_t>
+template <template <typename> class, typename>
+SpacePointContainer<container_t, holder_t>::SpacePointContainer(
+    const Acts::SpacePointContainerConfig& config,
+    const Acts::SpacePointContainerOptions& options, container_t&& container)
+    : m_config(config.toInternalUnits()),
+      m_options(options.toInternalUnits()),
+      m_container(std::move(container)) {
+  this->initialize();
```

Comment:
Is this practice in ACTS to use this keyword in such situation?

Reply:
We don't conventionally do that.